AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientExtensions.swift

```diff
@@ -940,8 +940,14 @@ extension AWSMobileClient {
             completionHandler(nil, AWSMobileClientError.notSignedIn(message: notSignedInErrorMessage))
             return
         }
-        let userDetails = AWSMobileClientUserDetails(with: self.userpoolOpsHelper.currentActiveUser!)
-        userDetails.getUserAttributes(completionHandler: completionHandler)
+        self.getTokens { _, error in
+            guard error == nil else {
+                completionHandler(nil, error)
+                return
+            }
+            let userDetails = AWSMobileClientUserDetails(with: self.userpoolOpsHelper.currentActiveUser!)
+            userDetails.getUserAttributes(completionHandler: completionHandler)
```

Comment:
Let's not force unwrap this.
```suggestion
            guard let currentActiveUser = self.userpoolOpsHelper.currentActiveUser else {
                let noActiveUserError = <some appropriate error>
                completionHandler(nil, noActiveUserError)
            }
            let userDetails = AWSMobileClientUserDetails(with: currentActiveUser)
```